File: naked-generator/Secondary/py/Secondary.py
from enum import Enum
from struct import pack, unpack
from collections import namedtuple


class Sync_State(Enum):
    MAX_START = 0
    MAX_END = 1
    MIN_START = 2
    MIN_END = 3


class Pedal(Enum):
    ACCELERATOR = 0
    BRAKE = 1
    ALL = 2


# SET_PEDALS_RANGE
class SET_PEDALS_RANGE:
    tuple = namedtuple("SET_PEDALS_RANGE", "sync_state, pedal", rename=True)
    schema = "<bb"
    
    @staticmethod
    def serialize(sync_state, pedal) -> bytes:
        return pack(SET_PEDALS_RANGE.schema, sync_state, pedal)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "SET_PEDALS_RANGE":
        return SET_PEDALS_RANGE.tuple._make(unpack(SET_PEDALS_RANGE.schema, buffer))


# ACCELERATOR_PEDAL_VAL
class ACCELERATOR_PEDAL_VAL:
    tuple = namedtuple("ACCELERATOR_PEDAL_VAL", "level", rename=True)
    schema = "<B"
    
    @staticmethod
    def serialize(level) -> bytes:
        return pack(ACCELERATOR_PEDAL_VAL.schema, level)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "ACCELERATOR_PEDAL_VAL":
        return ACCELERATOR_PEDAL_VAL.tuple._make(unpack(ACCELERATOR_PEDAL_VAL.schema, buffer))


# BRAKE_PEDAL_VAL
class BRAKE_PEDAL_VAL:
    tuple = namedtuple("BRAKE_PEDAL_VAL", "level", rename=True)
    schema = "<B"
    
    @staticmethod
    def serialize(level) -> bytes:
        return pack(BRAKE_PEDAL_VAL.schema, level)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "BRAKE_PEDAL_VAL":
        return BRAKE_PEDAL_VAL.tuple._make(unpack(BRAKE_PEDAL_VAL.schema, buffer))


# IMU_ANGULAR_RATE
class IMU_ANGULAR_RATE:
    tuple = namedtuple("IMU_ANGULAR_RATE", "ang_rate_x, ang_rate_y, ang_rate_z", rename=True)
    schema = "<HHH"
    
    @staticmethod
    def serialize(ang_rate_x, ang_rate_y, ang_rate_z) -> bytes:
        return pack(IMU_ANGULAR_RATE.schema, ang_rate_x, ang_rate_y, ang_rate_z)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IMU_ANGULAR_RATE":
        return IMU_ANGULAR_RATE.tuple._make(unpack(IMU_ANGULAR_RATE.schema, buffer))


# IMU_ACCELERATION
class IMU_ACCELERATION:
    tuple = namedtuple("IMU_ACCELERATION", "accel_x, accel_y, accel_z", rename=True)
    schema = "<HHH"
    
    @staticmethod
    def serialize(accel_x, accel_y, accel_z) -> bytes:
        return pack(IMU_ACCELERATION.schema, accel_x, accel_y, accel_z)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IMU_ACCELERATION":
        return IMU_ACCELERATION.tuple._make(unpack(IMU_ACCELERATION.schema, buffer))


# IRTS_FL_0
class IRTS_FL_0:
    tuple = namedtuple("IRTS_FL_0", "channel1, channel2, channel3, channel4", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel1, channel2, channel3, channel4) -> bytes:
        return pack(IRTS_FL_0.schema, channel1, channel2, channel3, channel4)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FL_0":
        return IRTS_FL_0.tuple._make(unpack(IRTS_FL_0.schema, buffer))


# IRTS_FL_1
class IRTS_FL_1:
    tuple = namedtuple("IRTS_FL_1", "channel5, channel6, channel7, channel8", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel5, channel6, channel7, channel8) -> bytes:
        return pack(IRTS_FL_1.schema, channel5, channel6, channel7, channel8)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FL_1":
        return IRTS_FL_1.tuple._make(unpack(IRTS_FL_1.schema, buffer))


# IRTS_FL_2
class IRTS_FL_2:
    tuple = namedtuple("IRTS_FL_2", "channel9, channel10, channel11, channel12", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel9, channel10, channel11, channel12) -> bytes:
        return pack(IRTS_FL_2.schema, channel9, channel10, channel11, channel12)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FL_2":
        return IRTS_FL_2.tuple._make(unpack(IRTS_FL_2.schema, buffer))


# IRTS_FL_3
class IRTS_FL_3:
    tuple = namedtuple("IRTS_FL_3", "channel13, channel14, channel15, channel16", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel13, channel14, channel15, channel16) -> bytes:
        return pack(IRTS_FL_3.schema, channel13, channel14, channel15, channel16)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FL_3":
        return IRTS_FL_3.tuple._make(unpack(IRTS_FL_3.schema, buffer))


# IRTS_FR_0
class IRTS_FR_0:
    tuple = namedtuple("IRTS_FR_0", "channel1, channel2, channel3, channel4", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel1, channel2, channel3, channel4) -> bytes:
        return pack(IRTS_FR_0.schema, channel1, channel2, channel3, channel4)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FR_0":
        return IRTS_FR_0.tuple._make(unpack(IRTS_FR_0.schema, buffer))


# IRTS_FR_1
class IRTS_FR_1:
    tuple = namedtuple("IRTS_FR_1", "channel5, channel6, channel7, channel8", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel5, channel6, channel7, channel8) -> bytes:
        return pack(IRTS_FR_1.schema, channel5, channel6, channel7, channel8)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FR_1":
        return IRTS_FR_1.tuple._make(unpack(IRTS_FR_1.schema, buffer))


# IRTS_FR_2
class IRTS_FR_2:
    tuple = namedtuple("IRTS_FR_2", "channel9, channel10, channel11, channel12", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel9, channel10, channel11, channel12) -> bytes:
        return pack(IRTS_FR_2.schema, channel9, channel10, channel11, channel12)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FR_2":
        return IRTS_FR_2.tuple._make(unpack(IRTS_FR_2.schema, buffer))


# IRTS_FR_3
class IRTS_FR_3:
    tuple = namedtuple("IRTS_FR_3", "channel13, channel14, channel15, channel16", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel13, channel14, channel15, channel16) -> bytes:
        return pack(IRTS_FR_3.schema, channel13, channel14, channel15, channel16)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_FR_3":
        return IRTS_FR_3.tuple._make(unpack(IRTS_FR_3.schema, buffer))


# IRTS_RL_0
class IRTS_RL_0:
    tuple = namedtuple("IRTS_RL_0", "channel1, channel2, channel3, channel4", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel1, channel2, channel3, channel4) -> bytes:
        return pack(IRTS_RL_0.schema, channel1, channel2, channel3, channel4)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RL_0":
        return IRTS_RL_0.tuple._make(unpack(IRTS_RL_0.schema, buffer))


# IRTS_RL_1
class IRTS_RL_1:
    tuple = namedtuple("IRTS_RL_1", "channel5, channel6, channel7, channel8", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel5, channel6, channel7, channel8) -> bytes:
        return pack(IRTS_RL_1.schema, channel5, channel6, channel7, channel8)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RL_1":
        return IRTS_RL_1.tuple._make(unpack(IRTS_RL_1.schema, buffer))


# IRTS_RL_2
class IRTS_RL_2:
    tuple = namedtuple("IRTS_RL_2", "channel9, channel10, channel11, channel12", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel9, channel10, channel11, channel12) -> bytes:
        return pack(IRTS_RL_2.schema, channel9, channel10, channel11, channel12)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RL_2":
        return IRTS_RL_2.tuple._make(unpack(IRTS_RL_2.schema, buffer))


# IRTS_RL_3
class IRTS_RL_3:
    tuple = namedtuple("IRTS_RL_3", "channel13, channel14, channel15, channel16", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel13, channel14, channel15, channel16) -> bytes:
        return pack(IRTS_RL_3.schema, channel13, channel14, channel15, channel16)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RL_3":
        return IRTS_RL_3.tuple._make(unpack(IRTS_RL_3.schema, buffer))


# IRTS_RR_0
class IRTS_RR_0:
    tuple = namedtuple("IRTS_RR_0", "channel1, channel2, channel3, channel4", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel1, channel2, channel3, channel4) -> bytes:
        return pack(IRTS_RR_0.schema, channel1, channel2, channel3, channel4)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RR_0":
        return IRTS_RR_0.tuple._make(unpack(IRTS_RR_0.schema, buffer))


# IRTS_RR_1
class IRTS_RR_1:
    tuple = namedtuple("IRTS_RR_1", "channel5, channel6, channel7, channel8", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel5, channel6, channel7, channel8) -> bytes:
        return pack(IRTS_RR_1.schema, channel5, channel6, channel7, channel8)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RR_1":
        return IRTS_RR_1.tuple._make(unpack(IRTS_RR_1.schema, buffer))


# IRTS_RR_2
class IRTS_RR_2:
    tuple = namedtuple("IRTS_RR_2", "channel9, channel10, channel11, channel12", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel9, channel10, channel11, channel12) -> bytes:
        return pack(IRTS_RR_2.schema, channel9, channel10, channel11, channel12)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RR_2":
        return IRTS_RR_2.tuple._make(unpack(IRTS_RR_2.schema, buffer))


# IRTS_RR_3
class IRTS_RR_3:
    tuple = namedtuple("IRTS_RR_3", "channel13, channel14, channel15, channel16", rename=True)
    schema = "<HHHH"
    
    @staticmethod
    def serialize(channel13, channel14, channel15, channel16) -> bytes:
        return pack(IRTS_RR_3.schema, channel13, channel14, channel15, channel16)
    
    @staticmethod
    def deserialize(buffer: bytes) -> "IRTS_RR_3":
        return IRTS_RR_3.tuple._make(unpack(IRTS_RR_3.schema, buffer))

File: naked-generator/Secondary/c/Secondary.c
#include <string.h>
#include <assert.h>
#include <stdio.h>

#include "Secondary.h"

/* Secondary_SET_PEDALS_RANGE */
void serialize_Secondary_SET_PEDALS_RANGE(Secondary_Sync_State sync_state, Secondary_Pedal pedal, uint8_t* buffer, size_t buf_len) {
    Secondary_SET_PEDALS_RANGE secondary_set_pedals_range = { sync_state, pedal };
	assert(buf_len >= sizeof(Secondary_SET_PEDALS_RANGE));
	memcpy(buffer, &secondary_set_pedals_range, sizeof(Secondary_SET_PEDALS_RANGE));
} 
void deserialize_Secondary_SET_PEDALS_RANGE(uint8_t* buffer, size_t buf_len, Secondary_SET_PEDALS_RANGE* secondary_set_pedals_range) {
	assert(buf_len >= sizeof(Secondary_SET_PEDALS_RANGE));
	memcpy(secondary_set_pedals_range, buffer, sizeof(Secondary_SET_PEDALS_RANGE));
}

/* Secondary_ACCELERATOR_PEDAL_VAL */
void serialize_Secondary_ACCELERATOR_PEDAL_VAL(uint8_t level, uint8_t* buffer, size_t buf_len) {
    Secondary_ACCELERATOR_PEDAL_VAL secondary_accelerator_pedal_val = { level };
	assert(buf_len >= sizeof(Secondary_ACCELERATOR_PEDAL_VAL));
	memcpy(buffer, &secondary_accelerator_pedal_val, sizeof(Secondary_ACCELERATOR_PEDAL_VAL));
} 
void deserialize_Secondary_ACCELERATOR_PEDAL_VAL(uint8_t* buffer, size_t buf_len, Secondary_ACCELERATOR_PEDAL_VAL* secondary_accelerator_pedal_val) {
	assert(buf_len >= sizeof(Secondary_ACCELERATOR_PEDAL_VAL));
	memcpy(secondary_accelerator_pedal_val, buffer, sizeof(Secondary_ACCELERATOR_PEDAL_VAL));
}

/* Secondary_BRAKE_PEDAL_VAL */
void serialize_Secondary_BRAKE_PEDAL_VAL(uint8_t level, uint8_t* buffer, size_t buf_len) {
    Secondary_BRAKE_PEDAL_VAL secondary_brake_pedal_val = { level };
	assert(buf_len >= sizeof(Secondary_BRAKE_PEDAL_VAL));
	memcpy(buffer, &secondary_brake_pedal_val, sizeof(Secondary_BRAKE_PEDAL_VAL));
} 
void deserialize_Secondary_BRAKE_PEDAL_VAL(uint8_t* buffer, size_t buf_len, Secondary_BRAKE_PEDAL_VAL* secondary_brake_pedal_val) {
	assert(buf_len >= sizeof(Secondary_BRAKE_PEDAL_VAL));
	memcpy(secondary_brake_pedal_val, buffer, sizeof(Secondary_BRAKE_PEDAL_VAL));
}

/* Secondary_IMU_ANGULAR_RATE */
void serialize_Secondary_IMU_ANGULAR_RATE(uint16_t ang_rate_x, uint16_t ang_rate_y, uint16_t ang_rate_z, uint8_t* buffer, size_t buf_len) {
    Secondary_IMU_ANGULAR_RATE secondary_imu_angular_rate = { ang_rate_x, ang_rate_y, ang_rate_z };
	assert(buf_len >= sizeof(Secondary_IMU_ANGULAR_RATE));
	memcpy(buffer, &secondary_imu_angular_rate, sizeof(Secondary_IMU_ANGULAR_RATE));
} 
void deserialize_Secondary_IMU_ANGULAR_RATE(uint8_t* buffer, size_t buf_len, Secondary_IMU_ANGULAR_RATE* secondary_imu_angular_rate) {
	assert(buf_len >= sizeof(Secondary_IMU_ANGULAR_RATE));
	memcpy(secondary_imu_angular_rate, buffer, sizeof(Secondary_IMU_ANGULAR_RATE));
}

/* Secondary_IMU_ACCELERATION */
void serialize_Secondary_IMU_ACCELERATION(uint16_t accel_x, uint16_t accel_y, uint16_t accel_z, uint8_t* buffer, size_t buf_len) {
    Secondary_IMU_ACCELERATION secondary_imu_acceleration = { accel_x, accel_y, accel_z };
	assert(buf_len >= sizeof(Secondary_IMU_ACCELERATION));
	memcpy(buffer, &secondary_imu_acceleration, sizeof(Secondary_IMU_ACCELERATION));
} 
void deserialize_Secondary_IMU_ACCELERATION(uint8_t* buffer, size_t buf_len, Secondary_IMU_ACCELERATION* secondary_imu_acceleration) {
	assert(buf_len >= sizeof(Secondary_IMU_ACCELERATION));
	memcpy(secondary_imu_acceleration, buffer, sizeof(Secondary_IMU_ACCELERATION));
}

/* Secondary_IRTS_FL_0 */
void serialize_Secondary_IRTS_FL_0(uint16_t channel1, uint16_t channel2, uint16_t channel3, uint16_t channel4, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FL_0 secondary_irts_fl_0 = { channel1, channel2, channel3, channel4 };
	assert(buf_len >= sizeof(Secondary_IRTS_FL_0));
	memcpy(buffer, &secondary_irts_fl_0, sizeof(Secondary_IRTS_FL_0));
} 
void deserialize_Secondary_IRTS_FL_0(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FL_0* secondary_irts_fl_0) {
	assert(buf_len >= sizeof(Secondary_IRTS_FL_0));
	memcpy(secondary_irts_fl_0, buffer, sizeof(Secondary_IRTS_FL_0));
}

/* Secondary_IRTS_FL_1 */
void serialize_Secondary_IRTS_FL_1(uint16_t channel5, uint16_t channel6, uint16_t channel7, uint16_t channel8, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FL_1 secondary_irts_fl_1 = { channel5, channel6, channel7, channel8 };
	assert(buf_len >= sizeof(Secondary_IRTS_FL_1));
	memcpy(buffer, &secondary_irts_fl_1, sizeof(Secondary_IRTS_FL_1));
} 
void deserialize_Secondary_IRTS_FL_1(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FL_1* secondary_irts_fl_1) {
	assert(buf_len >= sizeof(Secondary_IRTS_FL_1));
	memcpy(secondary_irts_fl_1, buffer, sizeof(Secondary_IRTS_FL_1));
}

/* Secondary_IRTS_FL_2 */
void serialize_Secondary_IRTS_FL_2(uint16_t channel9, uint16_t channel10, uint16_t channel11, uint16_t channel12, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FL_2 secondary_irts_fl_2 = { channel9, channel10, channel11, channel12 };
	assert(buf_len >= sizeof(Secondary_IRTS_FL_2));
	memcpy(buffer, &secondary_irts_fl_2, sizeof(Secondary_IRTS_FL_2));
} 
void deserialize_Secondary_IRTS_FL_2(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FL_2* secondary_irts_fl_2) {
	assert(buf_len >= sizeof(Secondary_IRTS_FL_2));
	memcpy(secondary_irts_fl_2, buffer, sizeof(Secondary_IRTS_FL_2));
}

/* Secondary_IRTS_FL_3 */
void serialize_Secondary_IRTS_FL_3(uint16_t channel13, uint16_t channel14, uint16_t channel15, uint16_t channel16, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FL_3 secondary_irts_fl_3 = { channel13, channel14, channel15, channel16 };
	assert(buf_len >= sizeof(Secondary_IRTS_FL_3));
	memcpy(buffer, &secondary_irts_fl_3, sizeof(Secondary_IRTS_FL_3));
} 
void deserialize_Secondary_IRTS_FL_3(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FL_3* secondary_irts_fl_3) {
	assert(buf_len >= sizeof(Secondary_IRTS_FL_3));
	memcpy(secondary_irts_fl_3, buffer, sizeof(Secondary_IRTS_FL_3));
}

/* Secondary_IRTS_FR_0 */
void serialize_Secondary_IRTS_FR_0(uint16_t channel1, uint16_t channel2, uint16_t channel3, uint16_t channel4, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FR_0 secondary_irts_fr_0 = { channel1, channel2, channel3, channel4 };
	assert(buf_len >= sizeof(Secondary_IRTS_FR_0));
	memcpy(buffer, &secondary_irts_fr_0, sizeof(Secondary_IRTS_FR_0));
} 
void deserialize_Secondary_IRTS_FR_0(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FR_0* secondary_irts_fr_0) {
	assert(buf_len >= sizeof(Secondary_IRTS_FR_0));
	memcpy(secondary_irts_fr_0, buffer, sizeof(Secondary_IRTS_FR_0));
}

/* Secondary_IRTS_FR_1 */
void serialize_Secondary_IRTS_FR_1(uint16_t channel5, uint16_t channel6, uint16_t channel7, uint16_t channel8, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FR_1 secondary_irts_fr_1 = { channel5, channel6, channel7, channel8 };
	assert(buf_len >= sizeof(Secondary_IRTS_FR_1));
	memcpy(buffer, &secondary_irts_fr_1, sizeof(Secondary_IRTS_FR_1));
} 
void deserialize_Secondary_IRTS_FR_1(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FR_1* secondary_irts_fr_1) {
	assert(buf_len >= sizeof(Secondary_IRTS_FR_1));
	memcpy(secondary_irts_fr_1, buffer, sizeof(Secondary_IRTS_FR_1));
}

/* Secondary_IRTS_FR_2 */
void serialize_Secondary_IRTS_FR_2(uint16_t channel9, uint16_t channel10, uint16_t channel11, uint16_t channel12, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FR_2 secondary_irts_fr_2 = { channel9, channel10, channel11, channel12 };
	assert(buf_len >= sizeof(Secondary_IRTS_FR_2));
	memcpy(buffer, &secondary_irts_fr_2, sizeof(Secondary_IRTS_FR_2));
} 
void deserialize_Secondary_IRTS_FR_2(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FR_2* secondary_irts_fr_2) {
	assert(buf_len >= sizeof(Secondary_IRTS_FR_2));
	memcpy(secondary_irts_fr_2, buffer, sizeof(Secondary_IRTS_FR_2));
}

/* Secondary_IRTS_FR_3 */
void serialize_Secondary_IRTS_FR_3(uint16_t channel13, uint16_t channel14, uint16_t channel15, uint16_t channel16, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_FR_3 secondary_irts_fr_3 = { channel13, channel14, channel15, channel16 };
	assert(buf_len >= sizeof(Secondary_IRTS_FR_3));
	memcpy(buffer, &secondary_irts_fr_3, sizeof(Secondary_IRTS_FR_3));
} 
void deserialize_Secondary_IRTS_FR_3(uint8_t* buffer, size_t buf_len, Secondary_IRTS_FR_3* secondary_irts_fr_3) {
	assert(buf_len >= sizeof(Secondary_IRTS_FR_3));
	memcpy(secondary_irts_fr_3, buffer, sizeof(Secondary_IRTS_FR_3));
}

/* Secondary_IRTS_RL_0 */
void serialize_Secondary_IRTS_RL_0(uint16_t channel1, uint16_t channel2, uint16_t channel3, uint16_t channel4, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RL_0 secondary_irts_rl_0 = { channel1, channel2, channel3, channel4 };
	assert(buf_len >= sizeof(Secondary_IRTS_RL_0));
	memcpy(buffer, &secondary_irts_rl_0, sizeof(Secondary_IRTS_RL_0));
} 
void deserialize_Secondary_IRTS_RL_0(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RL_0* secondary_irts_rl_0) {
	assert(buf_len >= sizeof(Secondary_IRTS_RL_0));
	memcpy(secondary_irts_rl_0, buffer, sizeof(Secondary_IRTS_RL_0));
}

/* Secondary_IRTS_RL_1 */
void serialize_Secondary_IRTS_RL_1(uint16_t channel5, uint16_t channel6, uint16_t channel7, uint16_t channel8, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RL_1 secondary_irts_rl_1 = { channel5, channel6, channel7, channel8 };
	assert(buf_len >= sizeof(Secondary_IRTS_RL_1));
	memcpy(buffer, &secondary_irts_rl_1, sizeof(Secondary_IRTS_RL_1));
} 
void deserialize_Secondary_IRTS_RL_1(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RL_1* secondary_irts_rl_1) {
	assert(buf_len >= sizeof(Secondary_IRTS_RL_1));
	memcpy(secondary_irts_rl_1, buffer, sizeof(Secondary_IRTS_RL_1));
}

/* Secondary_IRTS_RL_2 */
void serialize_Secondary_IRTS_RL_2(uint16_t channel9, uint16_t channel10, uint16_t channel11, uint16_t channel12, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RL_2 secondary_irts_rl_2 = { channel9, channel10, channel11, channel12 };
	assert(buf_len >= sizeof(Secondary_IRTS_RL_2));
	memcpy(buffer, &secondary_irts_rl_2, sizeof(Secondary_IRTS_RL_2));
} 
void deserialize_Secondary_IRTS_RL_2(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RL_2* secondary_irts_rl_2) {
	assert(buf_len >= sizeof(Secondary_IRTS_RL_2));
	memcpy(secondary_irts_rl_2, buffer, sizeof(Secondary_IRTS_RL_2));
}

/* Secondary_IRTS_RL_3 */
void serialize_Secondary_IRTS_RL_3(uint16_t channel13, uint16_t channel14, uint16_t channel15, uint16_t channel16, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RL_3 secondary_irts_rl_3 = { channel13, channel14, channel15, channel16 };
	assert(buf_len >= sizeof(Secondary_IRTS_RL_3));
	memcpy(buffer, &secondary_irts_rl_3, sizeof(Secondary_IRTS_RL_3));
} 
void deserialize_Secondary_IRTS_RL_3(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RL_3* secondary_irts_rl_3) {
	assert(buf_len >= sizeof(Secondary_IRTS_RL_3));
	memcpy(secondary_irts_rl_3, buffer, sizeof(Secondary_IRTS_RL_3));
}

/* Secondary_IRTS_RR_0 */
void serialize_Secondary_IRTS_RR_0(uint16_t channel1, uint16_t channel2, uint16_t channel3, uint16_t channel4, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RR_0 secondary_irts_rr_0 = { channel1, channel2, channel3, channel4 };
	assert(buf_len >= sizeof(Secondary_IRTS_RR_0));
	memcpy(buffer, &secondary_irts_rr_0, sizeof(Secondary_IRTS_RR_0));
} 
void deserialize_Secondary_IRTS_RR_0(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RR_0* secondary_irts_rr_0) {
	assert(buf_len >= sizeof(Secondary_IRTS_RR_0));
	memcpy(secondary_irts_rr_0, buffer, sizeof(Secondary_IRTS_RR_0));
}

/* Secondary_IRTS_RR_1 */
void serialize_Secondary_IRTS_RR_1(uint16_t channel5, uint16_t channel6, uint16_t channel7, uint16_t channel8, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RR_1 secondary_irts_rr_1 = { channel5, channel6, channel7, channel8 };
	assert(buf_len >= sizeof(Secondary_IRTS_RR_1));
	memcpy(buffer, &secondary_irts_rr_1, sizeof(Secondary_IRTS_RR_1));
} 
void deserialize_Secondary_IRTS_RR_1(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RR_1* secondary_irts_rr_1) {
	assert(buf_len >= sizeof(Secondary_IRTS_RR_1));
	memcpy(secondary_irts_rr_1, buffer, sizeof(Secondary_IRTS_RR_1));
}

/* Secondary_IRTS_RR_2 */
void serialize_Secondary_IRTS_RR_2(uint16_t channel9, uint16_t channel10, uint16_t channel11, uint16_t channel12, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RR_2 secondary_irts_rr_2 = { channel9, channel10, channel11, channel12 };
	assert(buf_len >= sizeof(Secondary_IRTS_RR_2));
	memcpy(buffer, &secondary_irts_rr_2, sizeof(Secondary_IRTS_RR_2));
} 
void deserialize_Secondary_IRTS_RR_2(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RR_2* secondary_irts_rr_2) {
	assert(buf_len >= sizeof(Secondary_IRTS_RR_2));
	memcpy(secondary_irts_rr_2, buffer, sizeof(Secondary_IRTS_RR_2));
}

/* Secondary_IRTS_RR_3 */
void serialize_Secondary_IRTS_RR_3(uint16_t channel13, uint16_t channel14, uint16_t channel15, uint16_t channel16, uint8_t* buffer, size_t buf_len) {
    Secondary_IRTS_RR_3 secondary_irts_rr_3 = { channel13, channel14, channel15, channel16 };
	assert(buf_len >= sizeof(Secondary_IRTS_RR_3));
	memcpy(buffer, &secondary_irts_rr_3, sizeof(Secondary_IRTS_RR_3));
} 
void deserialize_Secondary_IRTS_RR_3(uint8_t* buffer, size_t buf_len, Secondary_IRTS_RR_3* secondary_irts_rr_3) {
	assert(buf_len >= sizeof(Secondary_IRTS_RR_3));
	memcpy(secondary_irts_rr_3, buffer, sizeof(Secondary_IRTS_RR_3));
}
